sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +354,144 @@ export class AssessmentSequentialProcessingService {
+      .where("assessment.id = :assessmentId", { assessmentId: assessment.id })
+      .where("application.applicationNumber = :applicationNumber", {
+        applicationNumber: assessment.application.applicationNumber,
+      })
+      .orderBy("assessment.assessmentDate")
+      .getOne();
+    if (firstAssessment) {
+      firstAssessmentDate = getISODateOnlyString(
+        firstAssessment.assessmentDate,
+      );
+    }
+    const sfasApplicationLegacyRepo =
+      sfasApplicationType === "sfasPTApplication"
+        ? this.sfasPartTimeApplicationsRepo
+        : this.sfasApplicationsRepo;
+    const sfasApplicationLegacyAwards = sfasApplicationLegacyRepo
+      .createQueryBuilder(`${sfasApplicationType}`)
+      .select(`SUM(${sfasApplicationType}.csgp_award)`, "CSGP")
+      .addSelect(`SUM(${sfasApplicationType}.sbsd_award)`, "SBSD")
+      .addSelect(`SUM(${sfasApplicationType}.csgd_award)`, "CSGD")
+      .addSelect(`SUM(${sfasApplicationType}.bcag_award)`, "BCAG");
+    if (sfasApplicationType === "sfasPTApplication")
+      sfasApplicationLegacyAwards.addSelect(
+        `SUM(${sfasApplicationType}.cspt_award)`,
+        "CSPT",
+      );
+    sfasApplicationLegacyAwards
+      .innerJoin(`${sfasApplicationType}.individual`, "sfasStudent")
+      .where("lower(sfasStudent.lastName) = lower(:lastName)", { lastName })
+      .andWhere("sfasStudent.sin = :sin", { sin })
+      .andWhere("sfasStudent.birthDate = :birthDate", { birthDate })
+      .andWhere(
+        new Brackets((qb) => {
+          qb.where(`${sfasApplicationType}.startDate >= :startDate`, {
+            startDate: programYearStartDate,
+          }).andWhere(`${sfasApplicationType}.endDate <= :endDate`, {
+            endDate: firstAssessmentDate,
+          });
+        }),
+      );
+    return sfasApplicationLegacyAwards.getRawOne<SFASApplicationAwards>();
+  }
+
+  /**
+   * Push SFAS/SAIL application awards to total awards list.
+   * @param totals Total awards array.
+   * @param sfasLegacyAwards SFAS/SAIL application awards data.
+   * @param offeringIntensity Offering intensity.
+   */
+  appendSFASAwardsToTotals(
+    totals: AwardTotal[],
+    sfasLegacyAwards: SFASApplicationAwards,
+    offeringIntensity: OfferingIntensity,
+  ) {
+    // Loop through sfasLegacyAwards's keys to find existing value in totals.
+    // Modify the existing value or push the new data into totals.
+    Object.keys(sfasLegacyAwards).forEach((key: SFASApplicationAwardsEnum) => {
+      const index = totals.findIndex(
+        (value) =>
+          value.offeringIntensity === offeringIntensity &&
+          value.valueCode === key,
+      );
+      const awardValue = +sfasLegacyAwards[key];
+      if (index > -1) {
+        totals[index] = {
+          ...totals[index],
+          total: +totals[index].total + awardValue,
+        };
+      }
+      // Only push to totals if awardValue is greater than zero.
+      else if (awardValue > 0) {
```

Comment:
I have some confusion here: Line 484 updates the total in all scenarios i.e. `awardValue` greater or lesser than 0 whereas here we are pushing again a new object to the array after checking the `awardValue` greater than 0.